pkg/apis/batch/types.go

```diff
@@ -51,6 +51,9 @@ const (
 	// to the pod, which don't count towards the backoff limit, according to the
 	// pod failure policy. When the annotation is absent zero is implied.
 	JobIndexIgnoredFailureCountAnnotation = labelPrefix + "job-index-ignored-failure-count"
+	// JobControllerName reserved value for the managedBy field for the built-in
+	// Job controller.
+	JobControllerName = "job-controller.k8s.io"
```

Comment:
discussion with Jordan via slack: `"kubernetes.io/job-controller"` please. This matches the IngressClass.spec.controller format uses the name part from https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/names/controller_names.go#L56

Reply:
@deads2k IIUC, we used to select `batch.kubernetes.io/` as a prefix like this: https://github.com/kubernetes/kubernetes/blob/1bff5705fab60d4a17b0f21adc26f41f677f7dbb/pkg/apis/batch/types.go#L26-L29

So, wouldn't we use `batch.kubernetes.io/` as well? If I'm missing anything, please let me know, thanks.

Reply:
This is prefix to scope labels or annotations. For labels and annotations we can reuse them for different API objects, so the scope is "batch", "endpointslice", etc.

With "kubernetes.io" we prefix the "job-controller" to indicate this is kubernetes controller, but the scope of the controller is entire k8s. We cannot have multiple "job-controller"s inside k8s.

Reply:
That makes sense. Thank you for this clarifications!